src/c_api/c_api.cc

```diff
@@ -316,6 +316,37 @@ XGB_DLL int XGDMatrixGetStrFeatureInfo(DMatrixHandle handle, const char *field,
   API_END();
 }
 
+XGB_DLL int XGDMatrixSetFeatureInfo(DMatrixHandle handle, const char *field,
```

Comment:
Why not use SetFloatInfo API functions?

Reply:
I don't want to fix on floating point.  Numpy uses 64 bits by default, for that ctypes will make a copy.  It doesn't matter for feature weights, but I believe a generic interface is something we should have.